### PR TITLE
bazel/coverage: Test remotely

### PR DESCRIPTION
**Pull request by phlax**
- Use RBE for coverage tests
- Switch coverage VMs to ms-hosted small

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -73,6 +73,9 @@ stages:
 # Presubmit/default
 - ${{ if eq(variables.pipelineDefault, true) }}:
   - template: stages.yml
+    parameters:
+      checkStageDeps:
+      - env
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:78]: rm

File: .bazelrc [hunk 2]
```diff
@@ -13,6 +13,8 @@ startup --host_jvm_args=-Xmx3g
 fetch --color=yes
 run --color=yes
 
+build --action_env=ARB=v1 --host_action_env=ARB=v1
```
Comment (phlax) [inline .bazelrc:16]: rm

File: .bazelrc [hunk 3]
```diff
@@ -204,14 +204,19 @@ build:coverage --define=ENVOY_CONFIG_COVERAGE=1
...
+build:coverage --experimental_split_coverage_postprocessing
+build:coverage --experimental_fetch_all_coverage_outputs
 build:coverage --collect_code_coverage
 build:coverage --instrumentation_filter="^//source(?!/common/quic/platform)[/:],^//envoy[/:],^//contrib(?!/.*/test)[/:]"
-build:coverage --remote_download_toplevel
+build:coverage --remote_download_minimal
 build:coverage --define=tcmalloc=gperftools
+build:coverage --define=no_debug_info=1
+build:coverage --linkopt=-Wl,-s
+build:coverage --test_env=ENVOY_IP_TEST_VERSIONS=v4only
```
Comment (phlax) [inline .bazelrc:219]: these will affect mobile - not sure if that is correct - we can move

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**